galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+                               tabulatePSF().
+    @returns A dict containing the GSObject representing the PSF, where the keys are the bandpasses,
+    and the values are lists containing the PSF for that bandpass in each SCA.  Any entries for
+    which no PSF image was stored will be None.
+    """
+    # Get the image data.
+    im_list = galsim.fits.readMulti(image_filename)
+
+    # Get the metadata from the FITS table, and close the file when done.
+    hdu, hdu_list, fin = galsim.fits.readFile(data_filename)
+    bp_list = list(hdu_list[1].data.bandpass)
+    SCA_list = list(hdu_list[1].data.SCA)
+    galsim.fits.closeHDUList(hdu_list, fin)
+
+    # Set up the dict of PSF objects, indexed by bandpass (and then SCA).
+    PSF_dict = {}
+    for band_name in default_bandpass_list:
+        PSF_list = []
+
+        # Find all indices in `bp_list` that correspond to this bandpass.
+        bp_indices = []
+        if band_name in bp_list:
+            idx = -1
+            while True:
+                try:
+                    idx = bp_list.index(band_name, idx+1)
+                    bp_indices.append(idx)
+                except ValueError:
+                    break
+
+        for SCA in range(galsim.wfirst.n_sca+1):
+            if SCA == 0 or len(bp_indices)==0:
+                PSF_list.append(None)
+                continue
+
+            # Now find which element has both the right band_name and is for this SCA.  There might
+            # not be any, depending on what exactly was stored.
+            use_idx = -1
+            for index in bp_indices:
+                if SCA_list[index] == SCA:
+                    use_idx = index
+                    break
+            if use_idx<0:
+                PSF_list.append(None)
+                continue
+
+            # Now we know which PSF image is the right one.  So we should just make an
+            # InterpolatedImage out of it.
+            PSF = galsim.InterpolatedImage(im_list[use_idx])
+            PSF_list.append(PSF)
+
+        PSF_dict[band_name] = PSF_list
+
+    return PSF_dict
+
+def _read_aberrations(SCA):
+    """
+    This is a helper routine that reads in aberrations for a particular SCA and wavelength from
+    stored files.  It returns the aberrations in a format required by ChromaticOpticalPSF.
+
+    @param  SCA      The identifier for the SCA, from 1-9.
+    @returns a NumPy array containing the aberrations, in the required format for
+    ChromaticOpticalPSF.
+    """
+    if SCA < 1 or SCA > 9:
+        raise ValueError("SCA requested is out of range: %d"%SCA)
+
+    # Construct filename.
+    sca_str = '%02d'%SCA
+    infile = zemax_filepref + sca_str + zemax_filesuff
+
+    # Read in data.
+    dat = np.loadtxt(infile, skiprows=41, usecols=(2,)).transpose()
+    # Put it in the required format: an array of length 12, with the first entry empty (Zernike
+    # polynomials are 1-indexed so we use entries 1-11).  The units are waves.
```

Comment:
I guarantee we don't care about the piston term.  That only changes the phase of the light, which we don't care about.

I think we also should probably not include the tip/tilt terms.  The only effect of these are to shift the centroid of the convolved image.  While PSFs can involve a centroid shift, I'm not sure that pure tip/tilt terms are part of what users would typically consider the PSF.